### DISCUSSION TITLE
configuring gv/ghostview/display options in a remote/virtual environment

### DISCUSSION THREAD

**Discussion by fortworthjack** [dtcenter/METplus#1295] [File I/O] [answered]
Hi,

I'm going through this step, after ssh-ing into an AWS setup provided through the METplus training class that just started earlier this week.
https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-1-metplus-setupgrid-grid/met-tool-plot-data-plane/plot-data-plane-tool-plot-grib-data

Once in my AWS area, and what I assume to be the virtual machine within, after I make .ps output file, I'd kind of like to convert it to a .png file and then view it, but none of the view options work.

Is there anyway to view the file from the bowels of where it lives in that remote cloud and virtual environment?  If not, is there any way to export out the file to a place I can view it?

**Comment by hankenstein2** [comment]
Hi Jack, there should be a couple of ways to solve this. I *think* I just tried this out on your EC2 instance.

Could you try the simple convert <filename>.ps <filename>.png and tell me if that worked?

Alternately you could ssh into the EC2 instance with ssh -X and then run the gs command on the file which should display it on your computer.

I was able to get the file /d1/projects/METplus/METplus_Data/met_test/out/plot_data_plane/nam.t00z.awip1236.tm00.20070330_TMPZ2.ps to display on my laptop

Hank

**Reply by fortworthjack** [reply]
Hank,

I just ssh -X into my given IP address.  I did run a convert command to make a .png version of the wrfprs_TMP_2m.ps file.  So convert works. 

However, then running gv command on either .ps or .png file, I get:  "gv: Unable to open the display." 

echo $DISPLAY returns me
localhost:10.0

I tried to come in via a Putty.exe I have on my Windows machine, but it's no better.   If I didn't mention it, I'm doing the ssh -X connect command from a Windows cmd prompt.

**Comment by hankenstein2** [comment, marked as answer]
Running remote windows on Windows can be a bit tricky. Try following the
instructions in this web page and see if that helps.
You essentially need an extra piece of software to accomplish this.

http://www.geo.mtu.edu/geoschem/docs/putty_install.html

Hank

On Fri, Dec 3, 2021 at 2:29 PM Jack Settelmaier ***@***.***>
wrote:

> Hank,
>
> I just ssh -X into my given IP address. I did run a convert command to
> make a .png version of the wrfprs_TMP_2m.ps file. So convert works.
>
> However, then running gv command on either .ps or .png file, I get: "gv:
> Unable to open the display."
>
> echo $DISPLAY returns me
> localhost:10.0
>
> I tried to come in via a Putty.exe I have on my Windows machine, but it's
> no better. If I didn't mention it, I'm doing the ssh -X connect command
> from a Windows cmd prompt.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/1295#discussioncomment-1746785>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHAUIKGFNAGQNEGJKMIRD5DUPEZDNANCNFSM5JIHFH2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
--------------------------------------------------------------
Hank Fisher       Research Applications Laboratory
303-497-2817      National Center for Atmospheric Research
***@***.***  P.O. Box 3000, Boulder CO 80307-3000
--------------------------------------------------------------

**Reply by fortworthjack** [reply]
Thanks for this, I was poking around my Putty to see if I could solve this, and saw some X setting options.  I'm not sure, over VPN and such, this will all work, but I'll give it a try, to see if I can get anywhere to continue to follow along with the training.

**Reply by fortworthjack** [reply]
Ok, was able to follow those instructions, and got this close, hoping you can see the image?  What to try/fix next, please?

![image](https://user-images.githubusercontent.com/6979158/145078433-7fde1de5-526a-4453-9496-eec32776fd87.png)

**Reply by fortworthjack** [reply]
Actually, display -rotate 90 wrfprs.png works!!  So, I'm over the hump, I think, and I'm guessing "display" can serve my needs for visuals.

Though, at some point I'll wish I could move some of the input/output files (images, GRIB, etc.) elsewhere, to/from this AWS environment, if there is some FTP in/out option that can be used?  If/When I get to that point, perhaps I'll explore another Discussion post on that.

**Comment by hankenstein2** [comment]
Great to hear Jack, is it OK if I mark this discussion closed and you can
open another one for the file transfer off when you get there?

Hank

On Tue, Dec 7, 2021 at 10:42 AM Jack Settelmaier ***@***.***>
wrote:

> Actually, display -rotate 90 wrfprs.png works!! So, I'm over the hump, I
> think, and I'm guessing "display" can serve my needs for visuals.
>
> Though, at some point I'll wish I could move some of the input/output
> files (images, GRIB, etc.) elsewhere, to/from this AWS environment, if
> there is some FTP in/out option that can be used? If/When I get to that
> point, perhaps I'll explore another Discussion post on that.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/1295#discussioncomment-1767712>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHAUIKBLNPINFDGVN7VXEMTUPZBPZANCNFSM5JIHFH2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
--------------------------------------------------------------
Hank Fisher       Research Applications Laboratory
303-497-2817      National Center for Atmospheric Research
***@***.***  P.O. Box 3000, Boulder CO 80307-3000
--------------------------------------------------------------

**Reply by fortworthjack** [reply]
Yes, that works for me. Thanks!
